### PR TITLE
Feedback on task being assigned

### LINKED ISSUE #225: Feedback on task being assigned  during API call in AP
Whenever a card is dropped on another until we get a response show some sort of feedback on both the cards to the user and both the cards must be non-draggable during that period 
Reference :point_down: 

https://user-images.githubusercontent.com/45538392/141514064-543a69ef-6e72-4e31-97a5-26fad72db8c7.mp4

down:

### PR DESCRIPTION
**Pull request by akshay1502**
closes #225
Current drawbacks: We can only assign one task to members or vice-versa so as to see the Non-draggable behavior.

### CODE REVIEW

**Comment by akshay1502** [comment]
https://user-images.githubusercontent.com/55057874/145618789-6c981d7e-a18c-4d5b-b438-040c517eace1.mp4

File: src/components/availability-panel/drag-drop-context/DraggableComponent.tsx
Comment (swarajpure) [inline src/components/availability-panel/drag-drop-context/DraggableComponent.tsx:18]: than doing this, let's please use simple if else statements
lets not compromise on readability of code for length of code

File: src/components/availability-panel/drag-drop-context/DraggableComponent.tsx
Comment (swarajpure) [inline src/components/availability-panel/drag-drop-context/DraggableComponent.tsx:32]: logs in prod?

File: src/components/availability-panel/drag-drop-context/DraggableComponent.tsx
Comment (swarajpure) [inline src/components/availability-panel/drag-drop-context/DraggableComponent.tsx:30]: dead code?

File: src/components/availability-panel/drag-drop-context/DraggableComponent.tsx
Comment (swarajpure) [inline src/components/availability-panel/drag-drop-context/DraggableComponent.tsx:38]: let's not do the logical checks while we're calling the function, let's make it in some state or something and use that here instead

File: src/components/availability-panel/drag-drop-context/DraggableComponent.tsx
Comment (swarajpure) [inline src/components/availability-panel/drag-drop-context/DraggableComponent.tsx:61]: `e.target as HTMLImageElement`
seeing this syntax for the first time. Would like to more on about it, any resources for me?

File: src/components/availability-panel/drag-drop-context/DroppableComponent.tsx
Comment (swarajpure) [inline src/components/availability-panel/drag-drop-context/DroppableComponent.tsx:13]: re: dead code

File: src/components/availability-panel/drag-drop-context/DroppableComponent.tsx
Comment (swarajpure) [inline src/components/availability-panel/drag-drop-context/DroppableComponent.tsx:43]: again

File: src/components/availability-panel/drag-drop-context/index.tsx
Comment (swarajpure) [inline src/components/availability-panel/drag-drop-context/index.tsx:40]: what are these draggableId1 and 2? can you give more context so we can come up with better variable names

File: src/pages/availability-panel.tsx
Comment (swarajpure) [inline src/pages/availability-panel.tsx:37]: Please revert

File: src/pages/idle-members.tsx
Comment (swarajpure) [inline src/pages/idle-members.tsx:8]: please revert

File: src/components/availability-panel/drag-drop-context/DraggableComponent.tsx
Comment (akshay1502) [reply src/components/availability-panel/drag-drop-context/DraggableComponent.tsx:61]: https://www.designcise.com/web/tutorial/how-to-fix-property-does-not-exist-on-type-eventtarget-typescript-error
It is a typescript way of specifying event.target

File: src/components/availability-panel/drag-drop-context/index.tsx
Comment (akshay1502) [reply src/components/availability-panel/drag-drop-context/index.tsx:40]: draggableId1, draggableId2 are ids of task and member whenever we drop any task on member or vice-versa.

File: src/components/availability-panel/drag-drop-context/DraggableComponent.tsx
Comment (akshay1502) [reply src/components/availability-panel/drag-drop-context/DraggableComponent.tsx:40]: I will do this in the next commit, I thought of doing the same, but I remember it gave me some error.

File: src/components/availability-panel/drag-drop-context/DroppableComponent.tsx
Comment (akshay1502) [reply src/components/availability-panel/drag-drop-context/DroppableComponent.tsx]: I did in hurry, I didn't even see the commented code. sorry😅

File: src/components/availability-panel/drag-drop-context/index.tsx
Comment (swarajpure) [reply src/components/availability-panel/drag-drop-context/index.tsx:40]: Let's call them `taskId` and `memberId` in that case?

File: src/components/availability-panel/drag-drop-context/DraggableComponent.tsx
Comment (swarajpure) [reply src/components/availability-panel/drag-drop-context/DraggableComponent.tsx:63]: thanks!

**Comment by Pavangbhat** [comment]
@whyDontI please help us here how should we handle this. The idea is that once the card is dropped until we get the response we should not be able to do anything with that particular cards 

> Current drawbacks: We can only assign one task to members or vice-versa so as to see the undraggable behavior.

File: src/components/availability-panel/drag-drop-context/index.tsx
Comment (akshay1502) [reply src/components/availability-panel/drag-drop-context/index.tsx:40]: Actually we can't because we can't say which one is taskId and memberId since, we are dragging task over member or member over task.

Comment (swarajpure) [reply src/components/availability-panel/drag-drop-context/index.tsx:40]: I see, makes sense

File: src/components/availability-panel/drag-drop-context/index.tsx
Comment (swarajpure) [inline src/components/availability-panel/drag-drop-context/index.tsx:107]: Can we put line 103, 104 and 106, 107 in `finally` block?

**Comment by Pavangbhat** [comment]
Hi, @akshay1502  you told me that if we made changes in the backend then it works right?
I discussed the issue with Nikhil and he gave me an idea to solve the issue
The idea is that we can create a wrapper function and make a fetch request there and the wrapper function will return us the promise which will resolve and give us the data in the format that we want 
Something like this ⬇️ 
```
const assignTask = (taskId: string, userId: string) => {
  return new Promise(async (resolve, reject) => {
    try {
      const url = `${process.env.NEXT_PUBLIC_BASE_URL}/tasks/${taskId}`;
      const data = {
        status: ASSIGNED,
        assignee: userId,
      };

      const { requestPromise } = fetch({
        url,
        method: 'patch',
        data,
      });
      await requestPromise;
      resolve({
        taskId,
        userId,
      })
    } catch (error) {
      reject({
        error,
        taskId,
        userId,
      })
    }
  })
} 
```

**Comment by whyDontI** [comment]
to solve the `yarn.lock` conflicts just delete the `yarn.lock` file locally and run `yarn install`, this will generate the file again and solve conflicts.

**Comment by akshay1502** [comment]
https://user-images.githubusercontent.com/55057874/148018687-90795586-9fdc-44a0-bba8-e149e43ba33a.mp4

**Review by whyDontI — CHANGES_REQUESTED** [review]

**Review by whyDontI — COMMENTED** [review]
Added mocks to be used for development to emulate delay in response and different scenarios.

**Review by whyDontI — APPROVED** [review]
Good work @akshay1502 🥳 
I am approving and waiting for @Pavangbhat 's approval to merge this PR.

File: src/components/availability-panel/drag-drop-context/index.tsx
Comment (Pavangbhat) [inline src/components/availability-panel/drag-drop-context/index.tsx:95]: These are the changes of the new task workflow and must be included

**Review by Pavangbhat — CHANGES_REQUESTED** [review]
Some codes have been replaced which are necessary 
Please lets have them back

**Review by Pavangbhat — APPROVED** [review]
Everything looks good to me
Great work @akshay1502 🎉 🎉

**Pull request merged**
